warehouse/packaging/models.py

```diff
@@ -702,6 +702,18 @@ def trusted_published(self) -> bool:
             return False
         return all(file.uploaded_via_trusted_publisher for file in files)
 
+    @property
+    def is_source_verified(self) -> bool:
+        """
+        A Release can be considered source verified if it is trusted published
+        and the source URL is the same of the trusted publisher.
+        """
+        urls = self.urls
+        if not self.trusted_published or "Homepage" not in urls or urls["Homepage"] is None:
+            return False
+
+        return self.files.first().publisher_url == urls["Homepage"]
```

Comment:
Note that there could be multiple different publishers, not just one. We should iterate over all of `project.oidc_publishers`, make sure there is at least one event for a given publisher, and then compare `OIDCPublisher.publisher_url` with the URL we're trying to verify.

Reply:
This is also mixing up 'source' and 'homepage'. We also need to consider the case when the URL isn't exactly the same, but similar. The homepage "https://github.com/foo/bar/something.md" should still be verified if "https://github.com/foo/bar" is the publisher URL.